pingcap/br
# cmd,task: disable stray command line args, and log the command in LogArguments

### What problem does this PR solve? 

1. It was confusing that `--checksum false` does not actually disable checksum, because `--checksum` as a boolean flag already means `--checksum=true`. So `--checksum false` becomes `--checksum=true false`, and that extra `false` becomes a stray argument, which is ignored before this PR.

2. `LogArguments` did not record whether it is a "backup" or "restore" making it a bit hard to determine the data direction when reading logs.

### What is changed and how it works?

1. Added `Args: cobra.NoArgs` to all leaf commands so using `--checksum false` will result in

    ```
    Error: unknown command "false" for "br backup full"
    ```

    rather than silently keeping checksum.

2. Record the command being executed inside `LogArguments`.

    Note that I changed `LogArguments` from using `VisitAll` to `Visit`. Originally in #55 it was using `VisitAll` to print all args, however it was changed in #292 to condition on `f.Changed`, so perhaps let's just use `Visit` directly 🙃.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    * Run and check the log.

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - BR no longer accepts `--checksum false` in command line, which did not disable checksum. The correct usage always include the `=` sign: `--checksum=false`.
